src/trans.rs

```diff
@@ -160,30 +194,33 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
             }
         };
 
-        // Create the wasm locals
-        let mut locals = Vec::new();
+        // Create the wasm vars.
+        // Params and vars form the list of locals, both sharing the same index space.
+        let mut vars = Vec::new();
 
-        for mir_local in &self.mir.var_decls {
-            locals.push(rust_ty_to_binaryen(mir_local.ty));
+        for mir_var in &self.mir.var_decls {
+            vars.push(rust_ty_to_binaryen(mir_var.ty));
         }
 
-        for mir_local in &self.mir.temp_decls {
-            locals.push(rust_ty_to_binaryen(mir_local.ty));
+        for mir_var in &self.mir.temp_decls {
+            vars.push(rust_ty_to_binaryen(mir_var.ty));
         }
 
-        if needs_ret_local {
-            locals.push(binaryen_ret);
+        if needs_ret_var {
+            vars.push(binaryen_ret);
         }
 
-        // Function prologue: stack pointer
-        locals.push(BinaryenInt32());
-        let stack_pointer_local = BinaryenIndex(locals.len() as _);
+        // Function prologue: stack pointer local
+        vars.push(BinaryenInt32());
+        let stack_pointer_local = BinaryenIndex((binaryen_args.len() + vars.len() - 1) as u32);
 
-        // relooper helper for irreducible control flow
-        locals.push(BinaryenInt32());
-        let relooper_local = BinaryenIndex(locals.len() as _);
+        // relooper helper local for irreducible control flow
+        vars.push(BinaryenInt32());
+        let relooper_local = BinaryenIndex((binaryen_args.len() + vars.len() - 1) as u32);
 
-        debug!("{} wasm locals found", locals.len());
+        let locals_count = binaryen_args.len() + vars.len();
+        debug!("{} wasm locals initially found - params: {}, vars: {} (incl. stack pointer helper ${}, relooper helper ${})",
```

Comment:
probably not, but yeah we'll rustfmt eventually.
With rust's rightward drift even on my 13-inch laptop I sometimes go to 130 without noticing, and probably more at work...